src/pages/iou/steps/MoneyRequestAmountPage.js

```diff
@@ -76,15 +87,47 @@ class MoneyRequestAmountPage extends React.Component {
     }
 
     componentDidMount() {
+        if (this.isEditing) {
+            if (_.isEmpty(this.props.moneyRequest.participants) || this.props.moneyRequest.amount === 0) {
+                Navigation.goBack();
+                Navigation.navigate(ROUTES.getMoneyRequestRoute(this.iouType, this.reportID));
+            }
+        } else {
+            // Set the money request participants based on the report participants
+            const participants = ReportUtils.isPolicyExpenseChat(this.props.report)
+                ? OptionsListUtils.getPolicyExpenseReportOptions(this.props.report)
+                : OptionsListUtils.getParticipantsOptions(this.props.report, this.props.personalDetails);
+            this.props.moneyRequest.setParticipants(_.map(participants, (participant) => ({...participant, selected: true})));
+        }
+
         this.focusTextInput();
 
         // Focus automatically after navigating back from currency selector
         this.unsubscribeNavFocus = this.props.navigation.addListener('focus', () => {
             this.focusTextInput();
-            this.getCurrencyFromRouteParams();
```

Comment:
Is there any reason why we had to move this logic from the focus callback?

Reply:
First of all, there is a missing changes which I just pushed.

So, I found a bug where the currency in initial amount page does not get updated if we change the currency in edit amount page.
1. Send money, choose currency A and go to the confirmation page.
2. Open the edit amount page and change the currency to B and press Save
3. Go back to the first page and notice the currency is still A instead of B

This is because we update the currency from the param value every time the page get focused which will override the new currency updated from edit amount page.

Reply:
The bug is related to outdated state and is fixed by checking if moneyRequest.currency got changed. Can you elaborate how the param is related to this? We will update from the param only if there is a value and it's not the case here.

Reply:
> We will update from the param only if there is a value

No, we only update the currency state from the param if there is a value AND the param value changes. So, we have this PR https://github.com/Expensify/App/pull/18995 that change how the currency selection works. Now, when we select a currency, it will navigate back with the selected currency. Let say I select currency A, then the navigation will be to /send/money/?currency=A. On press Next, we will save the selected currency in the context state. Then, we edit the currency in the edit amount page, let say to currency B. When we navigate back to the first page, the focus listener will be triggered and get the currency from the param, which is A and update it to the state.

Reply:
Ah right, I had to explicitly set a currency on first page. Please mark this as resolved.